### PR TITLE
Fix/deagle-weapon-skill-level

### PR DESCRIPTION
**Pull request by Jeroen52**
Fixed a bug where the deagle damage would be dropped.

Hopefully this is the correct floating point accuracy.

### CODE REVIEW

File: weapon-config.inc [hunk 1]
```diff
@@ -4520,6 +4520,18 @@ static ProcessDamage(&playerid, &issuerid, &Float:amount, &weaponid, &bodypart,
 			}
 		}
 	}
+	
+	// Check deagle damage
+	if (weaponid == WEAPON_CHAINSAW) {
```
Comment (oscar-broman) [inline weapon-config.inc]: so.. did you test this? :smile:

Comment (Jeroen52) [inline weapon-config.inc]: /facepalm
The weird thing was, it SOMEHOW worked.

Like, I made sure that the player's weapon skill was lower than 200 and they could damage me.

Also, I have accidentally reset my branch before where the progress went lost.

Comment (Jeroen52) [inline weapon-config.inc]: Anyways, I'll do some more testing today because this is an embarrassment and I brought it onto myself all on my own.

**Comment by Jeroen52** [comment]
Did some testing today, this is the output of the debug.
```
[19:52:32] (wc:1) OnPlayerWeaponShot(1 shot 2 with 24 at 0.073242, 0.091674, 0.324985)
[19:52:32] (wc:1) OnPlayerGiveDamage(1 gave 46.200000 to 2 using 24 on bodypart 3)
[19:52:32] (wc) InflictDamage(2, 40.0000, 1, 24, 3) length = 3.000742
[19:52:32] (wc) OnPlayerDamageDone(1 did 40.000000 to 2 with 24 on bodypart 3)
[19:52:32] (wc:2) OnPlayerTakeDamage(2 took 46.200000 from 1 by 24 on bodypart 3)
[19:52:37] (wc:2) OnPlayerWeaponShot(2 shot 1 with 24 at 0.027587, 0.125000, 0.407318)
[19:52:37] (wc:2) OnPlayerGiveDamage(2 gave 23.100000 to 1 using 24 on bodypart 3)
[19:52:37] (wc) InflictDamage(1, 40.0000, 2, 24, 3) length = 3.433871
[19:52:37] (wc) OnPlayerDamageDone(2 did 40.000000 to 1 with 24 on bodypart 3)
[19:52:37] (wc:1) OnPlayerTakeDamage(1 took 23.100000 from 2 by 24 on bodypart 3)
```

**Comment by Jeroen52** [comment]
After the whoopsie, I tested it some more and it looks like it works.

**Comment by oscar-broman** [comment]
:+1:

**Pull request merged**